### PR TITLE
fix: wrong router navigation when redirect to /uc

### LINKED ISSUE #5417: 新建的角色勾选 `禁止访问 Console` 后，`redirect_uri` 参数无效
### 系统信息

- 外部访问地址: http://halo.test.lxware.cn
- 启动时间: 2024-02-27 13:41
- 版本: 2.12.2
- 构建时间: 2024-02-05 15:14
- Git Commit: 46efa87
- Java: IBM Semeru Runtime Open Edition / 17.0.9+9
- 数据库: PostgreSQL / 16.1
- 操作系统: Linux / 5.15.0-91-generic
- 已激活主题: Daisyblocks
- 已启动插件:
  - [文章限制阅读 Pro](https://github.com/f2ccloud/plugin-restricted-reading-pro)
  - [企业市场插件](https://github.com/liuchangfitcloud/plugin-enterprise-market)
  - [图库管理](https://github.com/halo-sigs/plugin-photos)
  - [瞬间](https://github.com/halo-sigs/plugin-moments)
  - [Sitemap](https://github.com/halo-dev/plugin-sitemap)
  - [搜索组件](https://github.com/halo-dev/plugin-search-widget)
  - [RSS](https://github.com/halo-dev/plugin-feed)
  - [评论组件](https://github.com/halo-dev/plugin-comment-widget)
  - [应用市场](https://github.com/halo-dev/plugin-app-store)

### 使用的哪种方式运行？

Docker Compose

### 发生了什么？

当访问需要登录的页面（例如 `/uc/notifications`）时，系统会携带 `redirect_uri=原始地址` 跳转到登录页面，登录成功后再跳转到原始地址。

新建一个 `禁止访问 Console` 的角色，使用该角色下的用户进行上述操作，不会跳转到登录前的原始地址；取消勾选该选项后可以正常跳回到原始地址。
image

### 相关日志输出

_No response_

### 附加信息

_No response_

### PR DESCRIPTION
**Pull request by LIlGG**
#### What type of PR is this?

/kind bug
/area ui

#### What this PR does / why we need it:

修复当角色勾选禁止访问 Console 后，redirect_uri 参数无效的问题

#### How to test it?

测试角色禁止访问 console 时，当链接中携带 redirect_uri 参数，能否正常跳转

#### Which issue(s) this PR fixes:

Fixes #5417 

#### Does this PR introduce a user-facing change?
```release-note
修复当角色禁止访问 Console 后，redirect_uri 参数无效的问题
```

### CODE REVIEW

File: ui/console-src/router/guards/auth-check.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export function setupAuthCheckGuard(router: Router) {
         return;
       }
 
-      if (to.name === "whiteList") {
+      if (whiteList.includes(to.name)) {
```
Comment (ruibaby) [inline ui/console-src/router/guards/auth-check.ts:55]: Good catch！

**Review by ruibaby — APPROVED** [review]
/lgtm

**Review by guqing — COMMENTED** [review]
/lgtm

**Pull request merged**